crates/biome_json_parser/src/parser.rs

```diff
@@ -14,20 +14,28 @@ pub(crate) struct JsonParser<'source> {
 
 #[derive(Default, Debug, Clone, Copy)]
 pub struct JsonParserOptions {
-    pub allow_comments: bool,
-    pub allow_trailing_commas: bool,
+    pub allow_comments: Option<bool>,
+    pub allow_trailing_commas: Option<bool>,
```

Comment:
What do you think we keep this untouched? `JsonParserOptions` holds the final resolved options, so it shouldn't use `Option<...>` anymore. We should have explict values when using this struct. I also don't see the necessity, instead of adding complexity. Correct me if I'm wrong.

Reply:
By the time we create this struct, we haven't resolved overrides yet.

https://github.com/biomejs/biome/blob/0c253b66b27168770c8cc8b07c1d8a2e1b6aad87/crates/biome_service/src/file_handlers/json.rs#L178-L191

Reply:
Yeah, but here we're creating it, we don't need to check any `None`s in it?

Reply:
Yeah I guess so. I'll try it and see how it turns out.

Reply:
> It could make caching overrides more difficult.

Feel free to elaborate if it is a concern.